### PR TITLE
US914145: Add liveness + readiness checks to Job Service

### PR DESCRIPTION
**Pull request by caoimhelynch**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=914145

### CODE REVIEW

File: job-service-dropwizard/src/main/resources/config.yaml [hunk 1]
```diff
@@ -25,3 +25,47 @@ server:
...
+      type: ready
+      critical: true
+      initialState: false
+      schedule:
+        initialDelay: ${JOB_SERVICE_READINESS_INITIAL_DELAY_DURATION:-15s}
+        checkInterval: ${JOB_SERVICE_READINESS_CHECK_INTERVAL_DURATION:-60s}
+        downtimeInterval: ${JOB_SERVICE_READINESS_DOWNTIME_INTERVAL_DURATION:-60s}
+        failureAttempts: ${JOB_SERVICE_READINESS_FAILURE_ATTEMPTS:-3}
+        successAttempts: ${JOB_SERVICE_READINESS_SUCCESS_ATTEMPTS:-1}
+
```
Comment (rorytorneymf) [inline job-service-dropwizard/src/main/resources/config.yaml]: Too many newlines?

File: job-service-container/README.md [hunk 2]
```diff
@@ -18,3 +18,36 @@ This docker image contains the [Job Service](../job-service) hosted using the Dr
 
 ## Feature Testing
 The testing for Job Service is defined [here](../testcases)
+
+### Environment Variables
```
Comment (rorytorneymf) [inline job-service-container/README.md]:
```suggestion
## Environment Variables
```

File: job-service-dropwizard/src/main/resources/config.yaml (hunk 1 above)
Comment (caoimhelynch) [reply job-service-dropwizard/src/main/resources/config.yaml]: The newline is being strange, this is just with one newline but it doesn't seem to be picking it up correctly. Can take this off? Then there will just be no newline. Or if you have any suggestions on how to make the newline get picked up?

Comment (rorytorneymf) [reply job-service-dropwizard/src/main/resources/config.yaml]: Think it was a tab. fixed now

Comment (caoimhelynch) [reply job-service-dropwizard/src/main/resources/config.yaml]: Ah thanks very much

File: job-service-container-tests/src/test/java/com/hpe/caf/services/job/api/JobServiceIT.java [hunk 3]
```diff
@@ -243,9 +243,30 @@ public void testHealthCheck() throws NoSuchAlgorithmException, KeyManagementExce
...
-        assertEquals("true", responseJson.at("/queue/healthy").asText());
+
+        final ObjectMapper mapper = new ObjectMapper();
+        final JsonNode jsonArray = mapper.readTree(responseString);
+
+        boolean isDatabaseHealthy = false;
+        boolean isPingHealthy = false;
+        boolean isQueueHealthy = false;
+
+        for (JsonNode node : jsonArray) {
```
Comment (dermot-hardy) [inline job-service-container-tests/src/test/java/com/hpe/caf/services/job/api/JobServiceIT.java]:
```suggestion
        for (final JsonNode node : jsonArray) {
```

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**